### ISSUE TITLE
Update runscripts

### ISSUE THREAD

**Issue by binkley** [binkley/modern-java-practices#226]
Show variants in the runscripts depending on platform/libraries.

For example, Spring Boot will rewrite the basenamed jar to just `<name>-<version>.jar`, but vanilla Maven uses `<name>-<version>-jar-with-dependencies.jar`.

**Comment by boxleytw** [comment]
If you're on Mac, the default `/bin/bash` is **crufty**: version 3 (current is 5), and some constructs in the runscripts fall down.

Consider:
- Downgrade the syntax in the runscripts to work with Mac's old `/bin/bash'
- Add checks to the runscripts to fail, and recommend that Mac users use Homebrew:
```sh
if ((BASH_VERSINFO[0] < 4)) # Change to use plain Bourne shell test
then
    cat <<EOM >&2
$0: This script requires Bash 4.0 or higher.
If you are on Mac, the native /bin/bash is from 2006 or older; consider
installing a current Bash from Homebrew.
EOM
    exit 1
fi
```

First thoughts are to downgrade the runscript to work on 2006 shells provided by Mac.
Second thoughts are to complain: Apple does not like command line work habits.
Third thoughts ... perhaps the script should just work.

**Comment by boxleytw** [comment]
Fix with 4aaff13954c614958af5ee8ec72db0aac99e6af4
